Project/DAQ_System/Program/Src/Sensor/Accelerometer/LSM6DSOXAccelerometer.cpp

```diff
@@ -0,0 +1,176 @@
+        average_array[i] = acc_magnitude * 0.001;
+    }
+
+    double min_value = std::numeric_limits<double>::max();
+    double max_value = -1;
+
+    // Find the largest and smallest magnitudes
+    for(int j = 2; j < kSampleCount; j++) {
+        if(average_array[j] <= min_value) {
+            min_value = average_array[j];
+
+        } else if (average_array[j] > max_value) {
+            max_value = average_array[j];
+        }
+    }
+
+    // Check that the maximum deviation is below a certain threshold.
+    // If it is too large, the conversion factor will be set to a previously
+    // determined conversion factor
+    gravity_adjustment_conversion_factor_ = 0;
+
+    if (max_value - min_value < 0.09) {
+        double total_array = 0;
+	    for(int i = 0; i < kSampleCount; i++) {
+		    total_array += average_array[i];
+	    }
+
+	    gravity_adjustment_conversion_factor_ = total_array / kSampleCount;
+    } else {
+        // Testing found that ~1.028 is about 1G
+        gravity_adjustment_conversion_factor_ = 1.028;
+    }
+
+}
+
+void LSM6DSOX::SetODR(SensorConfiguration::ODR ODRValue){
+	static constexpr uint8_t kNumBytes = 2;
+    uint8_t commands[kNumBytes] = {0};
+    commands[0] = CTRL1_XL;
+    uint8_t currentRegisterValue[1] = {0};
+
+    // read the current register values for the CTRL_X register
+	HAL_I2C_Master_Transmit(&i2c_,ACC_ADDRESS, commands,1, HAL_MAX_DELAY);
+	HAL_I2C_Master_Receive(&i2c_,ACC_ADDRESS, currentRegisterValue,1, HAL_MAX_DELAY);
+
+	// set the ODR bits to the desired value without touching the other bits in the register
+	// ODRValue is shifter 4 bits to properly align the desired ODR bits to the proper bits in the register
+	commands[1] = (currentRegisterValue[0] & 0x0F) | ODRValue << 4;
+
+	// write to register with the desired bit values
+	HAL_I2C_Master_Transmit(&i2c_,ACC_ADDRESS, commands,kNumBytes, HAL_MAX_DELAY);
+
+}
+
+void LSM6DSOX::SetFSR(SensorConfiguration::FSR FSRValue){
+	static constexpr uint8_t kNumBytes = 2;
+    uint8_t commands[kNumBytes] = {0};
+    commands[0] = CTRL1_XL;
+    uint8_t currentRegisterValue[1] = {0};
+
+    // read the current register values for the CTRL1_XL register
+	HAL_I2C_Master_Transmit(&i2c_,ACC_ADDRESS, commands,1, HAL_MAX_DELAY);
+	HAL_I2C_Master_Receive(&i2c_,ACC_ADDRESS, currentRegisterValue,1, HAL_MAX_DELAY);
+
+	// set the FSR bits to the desired value without touching the other bits in the register
+	// FSRValue is shifted to properly align the FSR bits in to the register.
+	commands[1] = (currentRegisterValue[0] & 0xF3) | FSRValue << 2  ;
+
+	// write to register with the desired bit values
+	HAL_I2C_Master_Transmit(&i2c_,ACC_ADDRESS, commands,kNumBytes, HAL_MAX_DELAY);
+
+	// each FSR has a different sensitivity factor to convert raw data to G
+	switch (FSRValue){
```

Comment:
I realized that if I wanted to make FSR changeable I would need to change how the values are scaled. Because of that I needed to remove the linear regression line I used to map output code to m/s^2., So I now just use sensitivity factor instead(totally didn't forget about doing this the first time I was converting raw data to m/s^2).  In  actuality the change isn't that huge because what the sensitivity factor is basically just the slope of the linear regression approached that I did last time.